test/components/async-route.spec.js

```diff
@@ -0,0 +1,152 @@
+import AsyncRoute from '../../src/components/async-route.vue';
+import Loading from '../../src/components/loading.vue';
+import PageBody from '../../src/components/page/body.vue';
+import TestUtilIcon from '../util/components/icon.vue';
+import TestUtilP from '../util/components/p.vue';
+import TestUtilSpan from '../util/components/span.vue';
+import { loadedAsync, setLoader } from '../../src/util/async-components';
+import { mount } from '../util/lifecycle';
+import { wait, waitUntil } from '../util/util';
+
+const mountComponent = (propsData) => mount(AsyncRoute, {
+  propsData: {
+    props: {},
+    loading: 'tab',
+    k: '0',
+    ...propsData
+  }
+});
+
+describe('AsyncRoute', () => {
```

Comment:
`AsyncRoute` is passed a few props derived from the route: `componentName`, `props`, and `k`. If the route changes, `componentName` might change (but might not, if it's the same route just with different route params). `k` might change, but again might not, and `props` will almost certainly change. We test what happens if `componentName` changes, since there's a watcher involved and some complexity. We also test that the component re-renders if `k` changes. We don't test what happens if `props` changes, but it's only used with `v-bind`, which will update automatically after a change. I don't think a test is needed there, but I'm also happy to add one!